tests/integration/devfile/cmd_devfile_catalog_test.go

```diff
@@ -14,32 +14,32 @@ var _ = Describe("odo devfile catalog command tests", func() {
 	var project string
 	var context string
 	var currentWorkingDirectory string
+	var cliRunner helper.CliRunner
+
+	// Using program commmand according to cliRunner in devfile
+	if os.Getenv("KUBERNETES") == "true" {
+		cliRunner = helper.NewKubectlRunner("kubectl")
+	} else {
+		cliRunner = helper.NewOcRunner("oc")
+	}
 
 	// This is run after every Spec (It)
 	var _ = BeforeEach(func() {
 		SetDefaultEventuallyTimeout(10 * time.Minute)
 		context = helper.CreateNewContext()
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
-		if os.Getenv("KUBERNETES") == "true" {
-			homeDir := helper.GetUserHomeDir()
-			kubeConfigFile := helper.CopyKubeConfigFile(filepath.Join(homeDir, ".kube", "config"), filepath.Join(context, "config"))
-			project = helper.CreateRandNamespace(kubeConfigFile)
-		} else {
-			project = helper.CreateRandProject()
-		}
+		homeDir := helper.GetUserHomeDir()
+		helper.CopyKubeConfigFile(filepath.Join(homeDir, ".kube", "config"), filepath.Join(context, "config"))
```

Comment:
You should not just assume that the kubeconfig will be in a `$HOME/.kube/config`.
Kubeconfig can be elsewhere. You first need to check if `$KUBECONFIG` if set.
If `$KUBECONFIG` is already set use that, if not use the `$HOME/.kube/config`.

Reply:
> You should not just assume that the kubeconfig will be in a $HOME/.kube/config.
Kubeconfig can be elsewhere. You first need to check if $KUBECONFIG if set.
If $KUBECONFIG is already set use that, if not use the $HOME/.kube/config.

@kadel Yes, `KUBECONFIG` can be somewhere and we need to check if $KUBECONFIG is set. But in `helper.CopyKubeConfigFile` we are again setting up `KUBECONFIG` env var with the temporary config context which mean if we use `$KUBECONFIG` as an arg to `helper.CopyKubeConfigFile` there might occur some racing condition in parallel test run.

Like, If we are running tests on two nodes and first node sets `KUBECONFIG` to some temporary config file and second node comes up and passes `$KUBECONFIG` which can pick up the the previously set file. This will again ruin everything.

IMO we should handle `$HOME/.kube/config` and `$HOME/admin.kubeconfig` according to the CI type. I mean on OpenShift CI or travis CI.

Reply:
> Like, If we are running tests on two nodes and first node sets `KUBECONFIG` to some temporary config file and second node comes up and passes `$KUBECONFIG` which can pick up the the previously set file. This will again ruin everything.

Doesn't each node use a separate environment? The second node should not be able to pick up the previously set file.

Reply:
> Doesn't each node use a separate environment? The second node should not be able to pick up the previously set file.

Actually previously I tried using `$KUBECONFIG` by passing `os.Getenv("KUBECONFIG")` as an arg to `copyKubeConfigFile()` but there I found the racing condition with parallel tests run as mentioned in [comment](https://github.com/openshift/odo/pull/3041#discussion_r425198685) and it was picking up the previously set temporary file and error says `no such file exists`

AFAIU It takes the existing `$KUBECONFIG`.

Reply:
Used this scenario here https://github.com/openshift/odo/commit/7eef087768f567eae9ab887e5327b5c3e472c50f#diff-d3d0e9ea582e16c9f0d4edabc01c6247R39 and got the failure `no such file exists` and then changed the approach after researching a lot.